magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/update_MAGIC_database.py

```diff
@@ -0,0 +1,322 @@
+            combined_df[column] = combined_df[column].replace(r'^\s*$', np.nan, regex=True)
+            not_null_data = combined_df[column].dropna()
+            if not_null_data.empty:
+                continue  # Skip if all values are NaN
+
+            inferred_type = pd.api.types.infer_dtype(not_null_data, skipna=True)
+
+            if inferred_type == "boolean":
+                default_value = False
+            elif inferred_type == "integer":
+                default_value = 0
+            elif inferred_type == "floating":
+                default_value = 0.0
+            elif inferred_type == "string":
+                default_value = "NaN"
+            else:
+                continue
+
+            combined_df[column] = (
+                combined_df[column]
+                .fillna(default_value)
+                .astype(type(not_null_data.iloc[0]))
+            )
+
+        combined_df = combined_df.infer_objects()
+
+    return combined_df
+
+def main():
+
+    """Main function."""
+    
+    parser = argparse.ArgumentParser()
+    
+    date_min_default = "20191101"
+    current_datetime = datetime.now()
+    date_max_default = current_datetime.strftime("%Y%m%d")
+
+    parser.add_argument(
+        "--date-min",
+        "-m",
+        dest="date_min",
+        type=str,
+        default=date_min_default,
+        help="Start of the time interval (in LST convention, format YYYYMMDD)."
+    )
+
+    parser.add_argument(
+        "--date-max",
+        "-M",
+        dest="date_max",
+        type=str,
+        default=date_max_default,
+        help="End of the time interval (in LST convention, format YYYYMMDD)."
+    )
+
+    args = parser.parse_args()
+
+    config = resource_file("database_config.yaml")
+
+    with open(
+        config, "rb"
+    ) as bf:
+        config_dict = yaml.safe_load(bf)
+
+    df_path = config_dict["database_paths"]["MAGIC+LST1"]
+    df_key = config_dict["database_keys"]["MAGIC+LST1"]
+    df = pd.read_hdf(df_path, key=df_key)
+
+    #Set "" to generate a new database.
+    previous_database_path = config_dict["database_paths"]["MAGIC"]
+
+    tel_id = [1, 2]
+    roman_numerals = {
+            1: "I",
+            2: "II"
+    }
+
+    file_exists = os.path.exists(previous_database_path)
+
+    if file_exists:
+
+        new_h5_file_path = (
+            "/fefs/aswg/workspace/joanna.wojtowicz/data/Common_MAGIC_LST1_data_MAGIC_runs_subruns_UPDATED.h5"
+        )
+
+        date_min = args.date_min
+        date_max = args.date_max
+
+        print("Updating database...")
+
+        database = table_magic_runs(df, date_min, date_max)
+
+        for tel in tel_id:
+
+            tel_roman = roman_numerals[tel]
+            key = config_dict["database_keys"][f"MAGIC-{tel_roman}"]
+
+            DF = pd.read_hdf(
+                previous_database_path,
+                key=key,
+            )
+
+            updated_df = update_tables(database, DF, tel)
+            print(updated_df)
+
+        try:
```

Comment:
you can extract this try except part after the if because saving is the same for both if cases, you just need to use the same variable for output path

Reply:
you marked it with thumb up, but the code still has two separate writing blocks. 
But it is not a big problem, we can merge it as it is and prepare a refactoring PR later on